pkg/master/master.go

```diff
@@ -132,7 +132,9 @@ type Config struct {
 	// ExternalHost is the host name to use for external (public internet) facing URLs (e.g. Swagger)
 	ExternalHost string
 
-	// If nil, the first result from net.InterfaceAddrs will be used.
+	// PublicAddress is the IP address where members of the cluster (kubelet,
+	// kube-proxy, services, etc.) can reach the master.
+	// If nil or 0.0.0.0, the host's default interface will be used.
```

Comment:
Okay. I changed it back closer to what it was before. The apiserver will set advertise-address = bind-address if advertise-address was unset. Then master.go setDefaults() will do the util.ChooseHostInterface() dance as it does currently if advertise-address is still 0.0.0.0.